tasks/pools.yml

```diff
@@ -0,0 +1,25 @@
+---
+
+- name: Setup rsyslog pool configuration
+  template:
+    src: 'etc/rsyslog.d/pool.conf.j2'
+    dest: '/etc/rsyslog.d/{{ item.name }}.conf'
+    owner: '{{ item.owner | default("root") }}'
+    group: '{{ item.group | default("root") }}'
+    mode: '{{ item.mode | default("0644") }}'
+    backup: '{{ item.backup | default("yes")  }}'
+  with_items: [ '{{ rsyslog_pools }}' ]
+  notify:
+    - Reload rsyslogd
```

Comment:
You need to add
```
when: item.enabled is defined and item.enabled
```
To this task, otherwise disabled tasks will be added and removed in a loop.